### PR TITLE
hammer: osd: adjust scrub boundary to object without SnapSet

### PR DESCRIPTION
**Pull request by dzafman**

### CODE REVIEW

**Comment by dzafman** [comment]
The last commit https://github.com/ceph/ceph/pull/11310/commits/ab9a0c42c433936611b6c7b3d06a03658ff32e0b is a cherry-pick from jewel.  It is only needed by Jewel and Hammer.

**Comment by dzafman** [comment]
@athanatos Fix commit added based on first rados suite run.  Repeat of individual job passed with the fix.  New runs started.

File: src/osd/PG.cc [hunk 1]
```diff
@@ -4159,6 +4159,11 @@ void PG::chunky_scrub(ThreadPool::TPHandle &handle)
           return;
         }
 
+        // Account for http://tracker.ceph.com/issues/17491
+        if (!scrubber.primary_scrubmap.objects.empty() &&
+            scrubber.primary_scrubmap.objects.rbegin()->first == scrubber.end)
+          scrubber.primary_scrubmap.objects.erase(scrubber.end);
```
Comment (athanatos) [inline src/osd/PG.cc]: Yep!

**Comment by dzafman** [comment]
@athanatos I should in some way indicate that the last commit is a cherry-pick from the _jewel_ branch which is unusual.

**Comment by athanatos** [comment]
@dzafman Probably sufficient to add "From jewel commit: " in front of what -x usually adds.

**Comment by dzafman** [comment]
@liewegas @athanatos I'm satisfied with these 2 final runs validated this even with 1 dead job.

No failures
http://pulpito.ceph.com/dzafman-2016-10-07_13:51:45-rados-wip-zafman-testing-hammer-distro-basic-smithi

One dead job because space ran out with 3 of 6 out OSDs causing backfull_toofull in this run.
http://pulpito.ceph.com/dzafman-2016-10-07_13:43:59-rados-wip-zafman-testing-hammer-distro-basic-smithi

**Comment by dzafman** [comment]
@liewegas Ready for review and merge after master and jewel versions.

**Comment by athanatos** [comment]
@liewegas The workaround in this branch in your opinion should be adequate, right?
@dzafman Have we been running this through the suite?

**Comment by liewegas** [comment]
It's ugly, but I think it's adequate.

**Comment by dzafman** [comment]
@athanatos Tests have been run.  Previous comments show the exact run locations before I removed needs-test.

**Comment by dzafman** [comment]
It has been decided that this change will NOT be merged to Hammer.

**Pull request closed without merging**